codec/encoder/core/loongarch/dct_lasx.c

```diff
@@ -0,0 +1,224 @@
+            psrc10, iStride1_x2,
+            psrc10, iStride1_x3,
+            src0, src1, src2, src3);
+  DUP4_ARG2(__lasx_xvldx,
+            psrc11, iStride0,
+            psrc11, iStride2,
+            psrc11, iStride2_x2,
+            psrc11, iStride2_x3,
+            src4, src5, src6, src7);
+  DUP4_ARG2(__lasx_xvldx,
+            psrc20, iStride0,
+            psrc20, iStride1,
+            psrc20, iStride1_x2,
+            psrc20, iStride1_x3,
+            src8, src9, src10, src11);
+  DUP4_ARG2(__lasx_xvldx,
+            psrc21, iStride0,
+            psrc21, iStride2,
+            psrc21, iStride2_x2,
+            psrc21, iStride2_x3,
+            src12, src13, src14, src15);
+  DUP4_ARG2(__lasx_xvilvl_b,
+            src0, src4,
+            src1, src5,
+            src2, src6,
+            src3, src7,
+            src0, src1, src2, src3);
+  DUP4_ARG2(__lasx_xvilvl_b,
+            src8, src12,
+            src9, src13,
+            src10, src14,
+            src11, src15,
+            src8, src9, src10, src11);
+  DUP4_ARG2(__lasx_xvhsubw_hu_bu,
+            src0, src0,
+            src1, src1,
+            src2, src2,
+            src3, src3,
+            src0, src1, src2, src3);
+  DUP4_ARG2(__lasx_xvhsubw_hu_bu,
+            src8, src8,
+            src9, src9,
+            src10, src10,
+            src11, src11,
+            src8, src9, src10 ,src11);
+  LASX_TRANSPOSE8x8_H(src0, src1, src2, src3, src8, src9, src10, src11,
+                      src0, src1, src2, src3, src8, src9, src10, src11);
+  DUP4_ARG3(__lasx_xvpermi_q,
+            src8, src0, 0x20,
+            src9, src1, 0x20,
+            src10,src2, 0x20,
+            src11,src3, 0x20,
+            src0, src1, src2, src3);
+  CALC_TEMPS_AND_PDCT(src0, src1, src2, src3,
+                      dst0, dst1, dst2, dst3);
+  DUP4_ARG3(__lasx_xvpermi_q,
+            dst0, dst0, 0x31,
+            dst1, dst1, 0x31,
+            dst2, dst2, 0x31,
+            dst3, dst3, 0x31,
+            dst4, dst5, dst6, dst7);
+  LASX_TRANSPOSE8x8_H(dst0, dst1, dst2, dst3, dst4, dst5, dst6, dst7,
+                      dst0, dst1, dst2, dst3, dst4, dst5, dst6, dst7);
+  DUP4_ARG3(__lasx_xvpermi_q,
+            dst4, dst0, 0x20,
+            dst5, dst1, 0x20,
+            dst6, dst2, 0x20,
+            dst7, dst3, 0x20,
+            dst0, dst1, dst2, dst3);
+  CALC_TEMPS_AND_PDCT(dst0, dst1, dst2, dst3,
+                      dst0, dst1, dst2, dst3);
+  DUP2_ARG2(__lasx_xvpackev_d,
+            dst1, dst0,
+            dst3, dst2,
+            tmp0, tmp1);
+  DUP2_ARG2(__lasx_xvpackod_d,
+            dst1, dst0,
+            dst3, dst2,
+            tmp2, tmp3);
+  DUP2_ARG3(__lasx_xvpermi_q,
+            tmp1, tmp0, 0x20,
+            tmp3, tmp2, 0x20,
+            dst0, dst1);
+  DUP2_ARG3(__lasx_xvpermi_q,
+            tmp1, tmp0, 0x31,
+            tmp3, tmp2, 0x31,
+            dst2, dst3);
+  __lasx_xvst(dst0, pDct, 0);
+  __lasx_xvst(dst1, pDct + 16, 0);
```

Comment:
__lasx_xvst(dst1, pDct, 16);

Same below.

Reply:
will update in next push